src/deluge/hid/display/oled.cpp

```diff
@@ -434,90 +436,191 @@ void OLED::sendMainImage() {
 struct TextLineBreakdown {
 	char const* lines[TEXT_MAX_NUM_LINES];
 	uint8_t lineLengths[TEXT_MAX_NUM_LINES];
+	uint32_t lineWidths[TEXT_MAX_NUM_LINES];
 	int32_t numLines;
 	int32_t longestLineLength;
 	int32_t maxCharsPerLine;
+	int32_t longestLineWidth;
+	int32_t maxWidthPerLine;
 };
 
-void breakStringIntoLines(char const* text, TextLineBreakdown* textLineBreakdown) {
-	textLineBreakdown->numLines = 0;
-	textLineBreakdown->longestLineLength = 0;
-
-	char const* lineStart = text;
-	char const* wordStart = lineStart;
-	char const* lineEnd =
-	    lineStart
-	    + textLineBreakdown
-	          ->maxCharsPerLine; // Default to it being the max length - we'll only use this if no "spaces" found.
+/// checks if a character fits within line width based on width of characters of added so far
+/// + the width of the character we are trying to add
+/// if the character fits, this function will return:
+/// 1) updated line width in pixels
+/// 2) updated line length in characters
+/// 3) number of spacing in pixels added after the last character
+bool addCharacterToLine(char c, int32_t maxWidthPerLine, int32_t textHeight, int32_t& lineWidth, int32_t& lineLength,
+                        int32_t& charSpacing) {
+	// we're in a word, calculate width (in px) of the character in that word
+	int32_t charWidth = deluge::hid::display::OLED::popup.getCharWidthInPixels(c, textHeight);
+	// is the cumulative width (in px) for the line we're looking at less than the max width of that line?
+	if ((lineWidth + charWidth) <= maxWidthPerLine) {
+		// this character fits within the line's width so increment the size of this line
+		lineWidth += charWidth;
+		// add spacing after the character so we can see if next character will fit in the line
+		// if we can't fit next character we'll need to remove this extra spacing (see below)
+		charSpacing = deluge::hid::display::OLED::popup.getCharSpacingInPixels(c, false);
+		lineWidth += charSpacing;
+
+		// increment the number of characters in this line
+		lineLength++;
+
+		return true;
+	}
+	return false;
+}
 
-findNextSpace:
+/// used with breakStringIntoLines function below
+/// here we iterate through every character in the string until we reach a "break" in a word
+/// which can be due to a new line, a space, an underscore or the end of the string (0)
+/// returns the following:
+/// 1) the string remaining after the split point
+/// 2) updated textLineBreakdown struct
+/// 3) width of the line in pixels from start of the line up to the split point
+/// 4) returns number of characters from start of the line up to the split point
+/// 5) returns number of spacing in pixels added after the last character
+char const* findWordSplitPoint(char const* wordStart, int32_t maxWidthPerLine, int32_t textHeight, int32_t& lineWidth,
```

Comment:
nit: find line split point

Find word split point sounds like trying to split a word over multiple lines